core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/UniqueIDReadWriteLDAPUserStoreManager.java

```diff
@@ -2185,13 +2187,23 @@ protected void modifyUserInRole(String userNameDN, String groupRDN, int modifyTy
                 logger.debug("User: " + userNameDN + " was successfully " + "modified in LDAP group: " + groupRDN);
             }
         } catch (NamingException e) {
-            String errorMessage =
-                    "Error occurred while modifying user entry: " + userNameDN + " in LDAP role: " + groupRDN;
-            if (log.isDebugEnabled()) {
-                log.debug(errorMessage, e);
+            if (e instanceof NameAlreadyBoundException) {
```

Comment:
We can catch this `NameAlreadyBoundException` also in a separate catch block before `NamingException` as suggested above by @sumedhe

Reply:
fixed